src/libratbag.c

```diff
@@ -1194,6 +1196,12 @@ ratbag_resolution_set_active(struct ratbag_resolution *resolution)
 	struct ratbag_profile *profile = resolution->profile;
 	struct ratbag_resolution *res;
 
+	log_debug(profile->device->ratbag,"Setting resolution %d to active (libratbag)\n",resolution->index);
+	if (resolution->dpi_x == 0) { /* if one axis is zero both have to be zero */
```

Comment:
You need to add a `is_disabled` field to the resolution and check it here instead of checking if the dpi is 0.